lib/getAlgorithm.js

```diff
@@ -0,0 +1,21 @@
+'use strict'
+
+const parseSignatureHeader = require('./parseSignatureHeader')
```

Comment:
I'm requiring the `parseSignatureHeader` helper function here but should it actually be included as a top-level method in the options object?

Reply:
parseSignature header is in internal method for us, not every implementation will have the signature header. Sometimes the signature is stored elsewhere. We _could_ maybe refactor this to be a `parseSignature` method that can be overridden in the same way as the other configs and called under the hood via options.ParseSignature(request) but lets consider that later, for now this being internal is fine.